snowflake.go

```diff
@@ -101,40 +125,53 @@ func NewNode(node int64) (*Node, error) {
 	nodeShift = StepBits
 
 	return &Node{
-		time: 0,
+		last: 0,
 		node: node,
 		step: 0,
 	}, nil
 }
 
 // Generate creates and returns a unique snowflake ID
+// use GenerateID to generate snowflakes with error checking
 func (n *Node) Generate() ID {
+	id, _ := n.doGenerateID()
+	return id
+}
+
+// GenerateID attempts to create and return a unique snowflake ID
+// If backwards clock drift is observed, a BackwardsTimeError will be returned
+func (n *Node) GenerateID() (ID, error) {
+	return n.doGenerateID()
+}
 
+func (n *Node) doGenerateID() (ID, error) {
 	n.mu.Lock()
 
 	now := time.Now().UnixNano() / 1000000
 
-	if n.time == now {
+	if n.last == now {
 		n.step = (n.step + 1) & stepMask
 
 		if n.step == 0 {
-			for now <= n.time {
+			for now <= n.last {
 				now = time.Now().UnixNano() / 1000000
 			}
 		}
+	} else if now < n.last{
+		return 0, NewBackwardsTimeError(n.last - now)
```

Comment:
Actually, this will break current implementation by not returning the generated ID. Fixing...